### PR TITLE
Fix using WithSpan annotation in java6 class

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy [hunk 1]
```diff
@@ -370,4 +381,56 @@ class WithSpanInstrumentationTest extends AgentInstrumentationSpecification {
...
     }
   }
+
+  def "instrument java6 class"() {
+    setup:
+    /*
+     class GeneratedJava6TestClass implements Runnable {
+       @WithSpan
+       public void run() {
+       }
```
Comment (mateuszrzeszutek) [inline instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy:391]:
```suggestion
       public void run() {
         TraceUtils.runUnderTrace("intercept", {})
       }
```

**Review by kubawach — COMMENTED** [review]
LGTM. We could probably make the java6 test more readable by moving it to a separate test module with maxJavaVersionForTests set to 6.

File: instrumentation/opentelemetry-annotations-1.0/javaagent/opentelemetry-annotations-1.0-javaagent.gradle [hunk 2]
```diff
@@ -8,6 +8,7 @@ dependencies {
   compileOnly project(path: ':opentelemetry-ext-annotations-shaded-for-instrumenting', configuration: 'shadow')
 
   testImplementation deps.opentelemetryExtAnnotations
+  testImplementation "net.bytebuddy:byte-buddy-gradle-plugin:${versions.bytebuddy}"
```
Comment (mateuszrzeszutek) [inline instrumentation/opentelemetry-annotations-1.0/javaagent/opentelemetry-annotations-1.0-javaagent.gradle:11]: Why gradle plugin and not just byte buddy?

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/opentelemetry-annotations-1.0/javaagent/opentelemetry-annotations-1.0-javaagent.gradle (hunk 2 above)
Comment (laurit) [reply instrumentation/opentelemetry-annotations-1.0/javaagent/opentelemetry-annotations-1.0-javaagent.gradle]: copy-pasted it without paying attention, fixed.

File: instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy (hunk 1 above)
Comment (laurit) [reply instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy]: Fixed

**Comment by mateuszrzeszutek** [comment]
> We could probably make the java6 test more readable by moving it to a separate test module with maxJavaVersionForTests set to 6.

Unfortunately I don't think this'd work: tests JVM process runs with our javaagent, which does not support Java <8.

File: instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy [hunk 3]
```diff
@@ -370,4 +381,57 @@ class WithSpanInstrumentationTest extends AgentInstrumentationSpecification {
...
+        void intercept(@This Object o) {
+          TraceUtils.runUnderTrace("intercept", {})
+        }
+      }))
+      .visit(new MemberAttributeExtension.ForMethod()
+        .annotateMethod(AnnotationDescription.Builder.ofType(WithSpan).build())
+        .on(ElementMatchers.named("run")))
+      .make()
+      .load(getClass().getClassLoader())
+      .getLoaded()
```
Comment (trask) [inline instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy:410]: nice, I would have found a (random old) library that was compiled to Java 6 and tested instrumentation of that library, but this is better 👍

**Review by trask — APPROVED** [review]

**Pull request merged**